instrumentation/opentelemetry-instrumentation-urllib3/src/opentelemetry/instrumentation/urllib3/__init__.py

```diff
@@ -92,6 +92,18 @@ def response_hook(span, request, response):
 _RequestHookT = typing.Optional[
     typing.Callable[[Span, urllib3.connectionpool.HTTPConnectionPool], None]
 ]
+_ExtendedRequestHookT = typing.Optional[
```

Comment:
Why not add it to the original hook and make the new parameters optional so that calling the span with the original arguments won't fail (and so remain backwards compatible)?

Reply:
the problem is that the function itself is an argument.

so if I have an old code that pass a function with only two arguments as `request_hook`
and in the instrumentor I call this function with 4 arguments, it will raise TypeError exception